.buildkite/commands/unit-tests-android.sh

```diff
@@ -3,5 +3,33 @@
 echo "--- :npm: Install Node dependencies"
 npm ci --unsafe-perm --prefer-offline --no-audit --no-progress
 
-echo "--- :node: Lint and Unit Tests"
+SECTION='--- :node: Android Unit Tests'
+set +e
+echo "$SECTION"
 CHECK_CORRECTNESS=false CHECK_TESTS=true TEST_RN_PLATFORM=android ./bin/ci-checks-js.sh
+TESTS_EXIT_CODE=$?
+set -e
+
+REPORT_SECTION_NAME='🚦 Report Tests Status'
+if [[ $TESTS_EXIT_CODE -eq 0 ]]; then
+    echo "--- $REPORT_SECTION_NAME"
+    echo "Android tests passed. 🎉"
+else
+    echo "+++ $REPORT_SECTION_NAME"
+    echo "Android tests failed."
+
+    if ! command -v ruby ; then
+      echo 'Skipping test reporting because Ruby is not available on this machine.'
+      exit $TESTS_EXIT_CODE
+    fi
```

Comment:
I switched the PR back to draft because #6538 revealed that the reporting doesn't run due to the image where this run not having Ruby 🤦‍♂️ 

https://buildkite.com/automattic/gutenberg-mobile/builds/8250#018cfb60-6ebd-44a1-9a61-848a09a0a754/1009-1011